### PR TITLE
Upgrade specs2 version to the same used by Play 2.4.6

### PR DESCRIPTION
**Pull request by marcospereira**

### CODE REVIEW

File: project/Play2-ReactiveMongo.scala [hunk 1]
```diff
@@ -114,6 +115,13 @@ object ShellPrompt {
 object Play2ReactiveMongoBuild extends Build {
   import BuildSettings._
 
+  val specsVersion = "3.6"
+  val specs2Dependencies = Seq(
+    "specs2-core",
+    "specs2-junit",
+    "specs2-mock"
```
Comment (cchantep) [inline project/Play2-ReactiveMongo.scala]: I don't think the mock dep is needed

File: project/Play2-ReactiveMongo.scala [hunk 2]
```diff
@@ -122,7 +130,8 @@ object Play2ReactiveMongoBuild extends Build {
         "Sonatype snapshots" at "https://oss.sonatype.org/content/repositories/snapshots/",
         "Sonatype" at "http://oss.sonatype.org/content/groups/public/",
         "Typesafe repository releases" at "http://repo.typesafe.com/typesafe/releases/",
-        "Typesafe repository snapshots" at "http://repo.typesafe.com/typesafe/snapshots/"
+        "Typesafe repository snapshots" at "http://repo.typesafe.com/typesafe/snapshots/",
+        "scalaz-bintray" at "https://dl.bintray.com/scalaz/releases"
```
Comment (cchantep) [inline project/Play2-ReactiveMongo.scala]: Is it required?

File: src/test/scala/PlaySpec.scala [hunk 3]
```diff
@@ -54,9 +55,10 @@ object PlaySpec extends org.specs2.mutable.Specification {
         lazy val router = play.api.routing.Router.empty
       }
 
-      apiFromCustomCtx.reactiveMongoApi.database.map(_ => {}).
-        aka("database resolution") must beEqualTo({}).
-        await(Common.timeoutMillis)
```
Comment (cchantep) [inline src/test/scala/PlaySpec.scala]: I really prefer to use the `.await` form

Comment (marcospereira) [inline src/test/scala/PlaySpec.scala]: It is not available anymore.

File: project/Play2-ReactiveMongo.scala (hunk 2 above)
Comment (marcospereira) [inline project/Play2-ReactiveMongo.scala]: Yes, it is. :-(

Quoting [Play 2.4 migration guide](https://www.playframework.com/documentation/2.4.x/Migration24#Specs2-support-in-a-separate-module):

> Additionally, specs2 now requires scalaz-stream which isn’t available on maven central or any other repositories that sbt uses by default.

Comment (cchantep) [inline project/Play2-ReactiveMongo.scala]: We are using specs2 3.6.4 without that, with no issue

File: src/test/scala/PlaySpec.scala (hunk 3 above)
Comment (cchantep) [inline src/test/scala/PlaySpec.scala]: It's still available, but requires `org.specs2.concurrent.ExecutionEnv`

Comment (marcospereira) [inline src/test/scala/PlaySpec.scala]: Yeah, I just saw it here: http://notes.implicit.ly/post/116619383574/specs2-35

Comment (marcospereira) [inline src/test/scala/PlaySpec.scala]: Fixed. :-)

File: project/Play2-ReactiveMongo.scala (hunk 2 above)
Comment (marcospereira) [inline project/Play2-ReactiveMongo.scala]: Travis build failed to retrieve scalaz dependency without the resolver:

https://travis-ci.org/ReactiveMongo/Play-ReactiveMongo/builds/115431769#L1156

Comment (cchantep) [inline project/Play2-ReactiveMongo.scala]: Ok, so the resolver can be added, and it can also be rebased. Thanks

**Comment by marcospereira** [comment]
@cchantep rebased.

**Pull request merged**